streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -310,48 +301,30 @@ public void shouldRestoreTimestampedStoreWithConverter() {
     @Test
     public void shouldUnregisterChangelogsDuringClose() {
         final ProcessorStateManager stateMgr = getStateManager(Task.TaskType.ACTIVE);
-        reset(storeMetadata);
-        final StateStore store = EasyMock.createMock(StateStore.class);
-        expect(storeMetadata.changelogPartition()).andStubReturn(persistentStorePartition);
-        expect(storeMetadata.store()).andStubReturn(store);
-        expect(store.name()).andStubReturn(persistentStoreName);
-
-        context.uninitialize();
-        store.init((StateStoreContext) context, store);
-        replay(storeMetadata, context, store);
+        final StateStore store = mock(StateStore.class);
+        when(store.name()).thenReturn(persistentStoreName);
 
         stateMgr.registerStateStores(singletonList(store), context);
-        verify(context, store);
+        verify(context).uninitialize();
+        verify(store).init((StateStoreContext) context, store);
 
         stateMgr.registerStore(store, noopStateRestoreCallback, null);
         assertTrue(changelogReader.isPartitionRegistered(persistentStorePartition));
 
-        reset(store);
-        expect(store.name()).andStubReturn(persistentStoreName);
-        store.close();
```

Comment:
Adding `verify(store).close()` causes the test to fail, `store.close()` is not executed at this point.

Reply:
Then I do not understand why the test passed before. I suppose there is a mistake in the test setup.

Reply:
So if I use the `Mockito.reset` to simulate what the original `EasyMock` was doing, i.e.

```java
reset(store);
when(store.name()).thenReturn(persistentStoreName);
store.close();
```

The the test passes. Note that you told me to remove this beforehand (see https://github.com/apache/kafka/pull/12524#discussion_r968477415)

Reply:
I think we are misunderstanding each other. With after line 314, I meant after `stateMgr.close();`. It could be that I mixed up the line number. Currently the test does not pass (as you can see in the builds) because the verification `verify(store).close();` is before `stateMgr.close();`.
The test should look like this:
```
    public void shouldUnregisterChangelogsDuringClose() {
        final ProcessorStateManager stateMgr = getStateManager(Task.TaskType.ACTIVE);
        final StateStore store = mock(StateStore.class);
        when(store.name()).thenReturn(persistentStoreName);

        stateMgr.registerStateStores(singletonList(store), context);
        verify(context).uninitialize();
        verify(store).init((StateStoreContext) context, store);

        stateMgr.registerStore(store, noopStateRestoreCallback, null);
        assertTrue(changelogReader.isPartitionRegistered(persistentStorePartition));

        stateMgr.close();
        verify(store).close();
        assertFalse(changelogReader.isPartitionRegistered(persistentStorePartition));
    }
```

Reply:
Ah that makes much more sense, done and committed.